File: 병화/타겟넘버.js
function solution(numbers, target) {
    let answer = 0;

    dfs(0, 0);

    function dfs(index, sum) {
        // 배열의 마지막 index에 도달했을 때 target 값과 일치하는지 확인
        if (index === numbers.length) {
            if (sum === target) {
                answer++;
            }
            return;
        }

        // 배열의 index번째 값이 더해지는 경우
        dfs(index + 1, sum + numbers[index]);
        // 배열의 index번째 값이 빼지는 경우
        dfs(index + 1, sum - numbers[index]);
    }
    return answer;
}

File: 병화/타겟넘버.py
answer = 0


def dfs(index, sum, numbers, target):
    global answer
    if index == len(numbers):
        if sum == target:
            answer += 1
        return

    dfs(index + 1, sum + numbers[index], numbers, target)
    dfs(index + 1, sum - numbers[index], numbers, target)


def solution(numbers, target):
    global answer
    dfs(0, 0, numbers, target)

    return answer